use-cases/order-picking/src/main/java/org/acme/orderpicking/util/LogUtil.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.acme.orderpicking.util;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.acme.orderpicking.domain.Order;
+import org.acme.orderpicking.domain.OrderItem;
+import org.acme.orderpicking.domain.OrderPickingSolution;
+import org.acme.orderpicking.domain.Trolley;
+import org.acme.orderpicking.domain.TrolleyStep;
+import org.slf4j.Logger;
+
+/**
+ * Helper class, not relevant.
+ */
+public class LogUtil {
+
+    private LogUtil() {
+    }
+
+    public static void log(Logger logger, OrderPickingSolution solution) {
+        if (solution == null) {
```

Comment:
```suggestion
        if (!logger.isDebugEnabled()) {
            return;
        }
        if (solution == null) {
```
To avoid all of the hard work if nothing gets printed in the end.